### DISCUSSION TITLE
sha256 username and password is throwing Bad Request

### DISCUSSION THREAD

**Discussion by milbreyrigor** [bluenviron/mediamtx#2210] [Questions]
How should we properly hash the user name and password? I used the code below

 using var hash = SHA256.Create();
 var byteArray = hash.ComputeHash(Encoding.UTF8.GetBytes(value));
Convert.ToBase64String(byteArray).ToLower()

But the resulting string prefixed with sha256: returns a Bad Request.

Even the sample in the API documentation also returns a bad request

**Comment by aler9** [comment]
Hashing of credentials works on the server-side, not on the client side. The client sends their credentials in plain (visible) format to the server. That's how it works in both RTSP and HTTP (entire internet).

**Comment by milbreyrigor** [comment]
Hi,

I may have missed out some information . I an using the API to create a
path configuration in the server side. I am trying to hass the readeruser
and readerpass properties of that path configuration.  Can we only do this
in plain text? Also, v1.0.0 does not seem to accept special characters e.g $

**Reply by aler9** [reply]
As i already wrote, credentials are sent in plain text in RTSP and every time you perform a login in any website.

Sending an encoded password to the server doesn't provide any additional security since it doesn't prevent MITM attacks.

Use RTSPS to encrypt credentials in transit and SHA256 to encrypt them in the configuration file.

**Comment by milbreyrigor** [comment]
I apologize for the confusion.

But I am not trying to access the RTSP stream.

What I am doing is using the api v2/config/paths/add:name to create a new
path in the configuration file.

For the path, I am try to set readUser and readPass as shown below. API doc
has mentioned this is how to do it. However, this would return a bad
request.

**Reply by aler9** [reply]
You are not providing any info that allows to replicate the problem, so this can go on forever. Open an issue and fill the issue template in order to provide data that allows to identify the issue.
